modules/image_processing/src/image_processor_service.cpp

```diff
@@ -21,20 +21,22 @@
 #include <image_processing/image_processor_service.h>
 
 kpsr_tutorial::ImageProcessingService::ImageProcessingService(kpsr::Environment * environment,
+                                                               int durationSleep,
              kpsr::Subscriber<kpsr::vision_ocv::ImageData> * imageDataSubscriber,
                                  kpsr::Publisher<ResultData> * resultPublisher)
     : kpsr::Service(environment, "image_processing")
+    ,_durationSleep(durationSleep)
     , _imageDataSubscriber(imageDataSubscriber)
     , _resultPublisher(resultPublisher)
 {}
 
 void kpsr_tutorial::ImageProcessingService::start() {
     std::function<void(const kpsr::vision_ocv::ImageData &)> listener = std::bind(&ImageProcessingService::onImageDataReceived, this, std::placeholders::_1);
-    _imageDataSubscriber->registerListener("image_processing", listener);
+    _imageDataSubscriber->registerListener("image_processing_"+ _resultPublisher->_publicationStats._name, listener);
```

Comment:
store this name in the private member so you can reuse it while removing it.

Reply:
Done